File: RainMusic/Views/Components/RainDesignSystem.swift
import SwiftUI

struct RainSpacing {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 12
    static let lg: CGFloat = 16
    static let xl: CGFloat = 20
    static let xxl: CGFloat = 24
    static let xxxl: CGFloat = 32
}

struct RainTypography {
    static let bodySmall: Font = .system(size: 12)
    static let body: Font = .system(size: 13)
    static let bodyMedium: Font = .system(size: 14)
    static let headline: Font = .system(size: 15)
    static let headlineLarge: Font = .system(size: 16)
    static let title: Font = .system(size: 28)
}

struct RainCornerRadius {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 6
    static let md: CGFloat = 8
    static let lg: CGFloat = 12
    static let xl: CGFloat = 16
    static let xxl: CGFloat = 20
    static let pill: CGFloat = 26
    static let full: CGFloat = 9999
}

struct RainShadow {
    static let small: Shadow = Shadow(color: .black.opacity(0.06), radius: 4, y: 2)
    static let medium: Shadow = Shadow(color: .black.opacity(0.12), radius: 8, y: 4)
    static let large: Shadow = Shadow(color: .black.opacity(0.15), radius: 10, y: 4)
    static let xl: Shadow = Shadow(color: .black.opacity(0.2), radius: 16, y: 8)
    
    static let accent: Shadow = Shadow(color: .rainAccent.opacity(0.4), radius: 10, y: 4)
    static let accentLarge: Shadow = Shadow(color: .rainAccent.opacity(0.3), radius: 16, y: 8)
}

struct RainLayout {
    static let appWidth: CGFloat = 390
    static let appHeight: CGFloat = 844
    static let statusBarHeight: CGFloat = 62
    static let bottomTabHeight: CGFloat = 95
    static let albumArtSize: CGFloat = 280
    static let progressBarWidth: CGFloat = 350
    static let controlsPanelHeight: CGFloat = 80
}

File: RainMusicAndroid/app/src/main/java/com/rain/music/ui/theme/RainDesignSystem.kt
package com.rain.music.ui.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

object RainSpacing {
    val xs = 4.dp
    val sm = 8.dp
    val md = 12.dp
    val lg = 16.dp
    val xl = 20.dp
    val xxl = 24.dp
    val xxxl = 32.dp
}

object RainTypography {
    const val bodySmall = 12.sp
    const val body = 13.sp
    const val bodyMedium = 14.sp
    const val headline = 15.sp
    const val headlineLarge = 16.sp
    const val title = 28.sp
}

object RainCornerRadius {
    val xs = RoundedCornerShape(4.dp)
    val sm = RoundedCornerShape(6.dp)
    val md = RoundedCornerShape(8.dp)
    val lg = RoundedCornerShape(12.dp)
    val xl = RoundedCornerShape(16.dp)
    val xxl = RoundedCornerShape(20.dp)
    val pill = RoundedCornerShape(26.dp)
    val full = RoundedCornerShape(9999.dp)
}

object RainShadow {
    val small = androidx.compose.ui.graphics.Shadow(Color.Black.copy(alpha = 0.06f), blurRadius = 4.dp, offsetY = 2.dp)
    val medium = androidx.compose.ui.graphics.Shadow(Color.Black.copy(alpha = 0.12f), blurRadius = 8.dp, offsetY = 4.dp)
    val large = androidx.compose.ui.graphics.Shadow(Color.Black.copy(alpha = 0.15f), blurRadius = 10.dp, offsetY = 4.dp)
    val xl = androidx.compose.ui.graphics.Shadow(Color.Black.copy(alpha = 0.2f), blurRadius = 16.dp, offsetY = 8.dp)
}

object RainLayout {
    val appWidth = 390.dp
    val appHeight = 844.dp
    val statusBarHeight = 62.dp
    val bottomTabHeight = 95.dp
    val albumArtSize = 280.dp
    val progressBarWidth = 350.dp
    val controlsPanelHeight = 80.dp
}